### ISSUE TITLE
Move expensive computation related to state sync / catch up into a separate actor

### ISSUE THREAD

**Issue by SkidanovAlex** [near/nearcore#3201]
Today calls to `combine_state_parts` take up to 20 seconds blocking `ClientActor`.
Similarly, calling to `catchup_blocks` can take a lot of time, since it in a single blocking call applies all the blocks downloaded while state for the next epoch had been downloading.

We want to move these two methods (or their callers) into a separate actor, so that neither blocks operation of the current epoch.

**Comment by pmnoxx** [comment]
@SkidanovAlex Thanks, I'll take a look

**Comment by pmnoxx** [comment]
I think I finally figured out how to split ClientActor.
https://github.com/near/nearcore/pull/3941/files

**Comment by EgorKulikov** [comment]
Main problem we have is that ClientActor behave as single thread event queue. When we process downloaded state parts and then catchup on blocks at the start of an epoch if we are going to be validator for the next epoch, in the meantime we would not process any blocks, which may be a problem if we are validating some other shard for this epoch

At the same time we cannot just put those tasks to some other thread/actor as they require some parts of Client, and those are not thread safe. There are 2 ways to overcome this:
1. Split work into small installments (applying one part of state/one block), and allow other tasks to get control of this client actor between those. Problem with this is that with relatively full blocks we may still not able to be fast enough to process blocks for current epoch after we've done piece of catchup
2. Introduce another actor that would do actual state part/block processing and send back message with results, upon receiving such message we'll then send message with next task.

Both solutions would require quite invasive changes to how we apply state part/catchup blocks

**Comment by bowenwang1996** [comment]
@EgorKulikov my understanding is that we want to go with the second approach. More specifically, we want to have the following:
* A separate actor that combines state parts in parallel. This does not interfere with the client actor and should be relatively straightforward to implement.
* A new actor that applies different chunks in multiple threads and sends the results back to the client actor. This change may be somewhat invasive.

cc @mzhangmzz @matklad

**Comment by bowenwang1996** [comment]
@EgorKulikov this is done right?

**Comment by bowenwang1996** [comment]
I believe this is already done. @EgorKulikov please reopen if it is not
